tests/accessor/accessor_common.h

```diff
@@ -1030,6 +1030,94 @@ decltype(auto) get_subscript_overload(const AccT& accessor, size_t index) {
   if constexpr (dims == 3) return accessor[index][index][index];
 }
 
+// Helper function to increment id linearly according to its linear position
+// id2+(id1*r2)+(id0*r1*r2) for dims = 3 or id1+(id0*r1) for dims = 2
+//
+// Incrementing index of last dimension if it not yet reached end of the range
+// otherwise reseting it and trying to increment index of previous dimension.
+template <int dims>
+void add_id_linear(sycl::id<dims>& id, size_t size) {
```

Comment:
I guess this function will not work any more if each range element is different, like the `{ 2, 3, 5 }` I was suggesting.
So, with your comment I understand what this function does. Thank. :-)
Then you could rename the function like
```suggestion
sycl::id<dims> next_id_linearly(const sycl::id<dims>& id, size_t size)
```
or
```suggestion
sycl::id<dims> next_lexicographic_id(const sycl::id<dims>& id, size_t size) {
```
and avoid to change the argument by side effect. Just return a new one.

Reply:
Changed to `sycl::id<dims> next_id_linearly` and parameter by value.